qiskit/primitives/containers/estimator_pub.py

```diff
@@ -132,6 +132,15 @@ def coerce(cls, pub: EstimatorPubLike, precision: float | None = None) -> Estima
                     validate=False,  # Assume Pub is already validated
                 )
             return pub
+
+        if isinstance(pub, QuantumCircuit):
+            raise ValueError(
+                f"An invalid Estimator pub-like was given ({type(pub)}). "
+                "If you want to run a single pub-like, you need to wrap it with `[]` like "
```

Comment:
I think I'd prefer reading "pub" than "pub-like" in the second sentence (it also avoids two likes in the sentence).  It's a tiny detail, but I think that the difference between pub and pub-like is a technicality and we can probably use them interchangeably. I expect both concepts to be foreign to the users anyway (pub, pub-like), but the example really clarifies the message, so I am not too set on this comment, either way is fine for me.

```suggestion
                "If you want to run a single pub, you need to wrap it with `[]` like "
```

Reply:
Yes, it makes sense to use just `pub` in the sentence.